src/font/lv_font_cache.c

```diff
@@ -0,0 +1,230 @@
+
+    /* value */
+    lv_draw_buf_t * draw_buf;
+} font_bitmap_cache_data_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static inline bool font_is_built_in(const lv_font_t * font);
+static void * get_bitmap_cached(lv_font_glyph_dsc_t * g_dsc, const lv_font_fmt_txt_dsc_t * fdsc, uint32_t gid);
+static bool font_bitmap_create_cb(font_bitmap_cache_data_t * data, void * user_data);
+static void font_bitmap_free_cb(font_bitmap_cache_data_t * data, void * user_data);
+static lv_cache_compare_res_t font_bitmap_compare_cb(const font_bitmap_cache_data_t * lhs,
+                                                     const font_bitmap_cache_data_t * rhs);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_font_cache_init(uint32_t max_glyph_cnt)
+{
+    LV_ASSERT_MSG(font_bitmap_cache == NULL, "Font cache already initialized");
+
+    lv_cache_ops_t ops = {
+        .compare_cb = (lv_cache_compare_cb_t)font_bitmap_compare_cb,
+        .create_cb = (lv_cache_create_cb_t)font_bitmap_create_cb,
+        .free_cb = (lv_cache_free_cb_t)font_bitmap_free_cb,
+    };
+
+    lv_cache_t * cache = lv_cache_create(&lv_cache_class_lru_rb_count, sizeof(font_bitmap_cache_data_t),
+                                         max_glyph_cnt, ops);
+    lv_cache_set_name(cache, "FONT_BITMAP");
+
+    font_bitmap_cache = cache;
+}
+
+void lv_font_cache_deinit(void)
+{
+    lv_cache_destroy(font_bitmap_cache, NULL);
+    font_bitmap_cache = NULL;
+}
+
+const void * lv_font_cache_get_glyph_bitmap(lv_font_glyph_dsc_t * g_dsc, lv_draw_buf_t * draw_buf)
+{
+    LV_ASSERT_NULL(g_dsc);
+
+    const lv_font_t * font = g_dsc->resolved_font;
+    if(!font_is_built_in(font)) {
+        /* If the font is not built-in, use the original function to get the bitmap */
+        return font->get_glyph_bitmap(g_dsc, draw_buf);
+    }
+
+    lv_font_fmt_txt_dsc_t * fdsc = (lv_font_fmt_txt_dsc_t *)font->dsc;
+    uint32_t gid = g_dsc->gid.index;
+    if(!gid) return NULL;
+
+    const lv_font_fmt_txt_glyph_dsc_t * gdsc = &fdsc->glyph_dsc[gid];
+    void * raw_glyph_bitmap = (void *)&fdsc->glyph_bitmap[gdsc->bitmap_index];
+
+    if(g_dsc->req_raw_bitmap) return raw_glyph_bitmap;
+
+    int32_t gsize = (int32_t) gdsc->box_w * gdsc->box_h;
+    if(gsize == 0) return NULL;
+
+    /* If the alignment, stride, and color format is correct, bypass the cache */
+    if(g_dsc->format == LV_FONT_GLYPH_FORMAT_A8
+       && raw_glyph_bitmap == lv_draw_buf_align(raw_glyph_bitmap, LV_COLOR_FORMAT_A8)
+       && g_dsc->box_w == lv_draw_buf_width_to_stride(g_dsc->box_w, LV_COLOR_FORMAT_A8)) {
+
+        const uint32_t a8_stride = gdsc->box_w * sizeof(uint8_t);
+
+        /* Use the static bitmap draw buffer to avoid memory allocation */
+        lv_draw_buf_init(
+            &font_static_bitmap_draw_buf,
```

Comment:
It can happen that 2 draw units are using the `font_static_bitmap_draw_buf` at the same time. 

IMO the user can always supply a `draw_buf *` to this function and we can initialize it here, or use it in `get_bitmap_cached` below. This way the life cycle of `draw_buf` managed on the user's side which is more flexible and transparent.

Reply:
Done, I added a check on `draw_buf`, when the `draw_buf` pointer is passed in externally, or the font cache is bypassed.
`font_static_bitmap_draw_buf` is only used in `LV_OS_NONE` mode, there will be no multithreading issues.